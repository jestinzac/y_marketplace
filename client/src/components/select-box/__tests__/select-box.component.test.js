import { render, screen, fireEvent } from "@testing-library/react";

import SelectBox from "../select-box.component";

describe("SelectBox", () => {
  let dataItems;
  beforeAll(() => {
    dataItems = [
      {
        id: 1,
        username: "John",
      },
      {
        id: 2,
        username: "Smith",
      },
    ];
  });

  it("should render the element without any values", () => {
    const { asFragment } = render(<SelectBox type="username" />);
    const options = screen.queryAllByTestId("option-value");

    expect(screen.getByTestId("username")).toHaveAttribute("id", "username");
    expect(screen.getByTestId("username")).toBeInTheDocument();
    expect(options.length).toBe(0);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the element with values if passed", () => {
    const { asFragment } = render(
      <SelectBox type="username" dataItems={dataItems} />
    );
    const options = screen.queryAllByTestId("option-value");

    expect(screen.getByTestId("username")).toHaveAttribute("id", "username");
    expect(screen.getByTestId("username")).toBeInTheDocument();
    expect(options.length).toBe(2);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the element with values and trigger event on change", async () => {
    const onChangeHandlerMock = jest.fn(() => "Change triggered");
    const { asFragment } = render(
      <SelectBox
        type="username"
        dataItems={dataItems}
        onChangeHandler={onChangeHandlerMock}
      />
    );
    const options = screen.queryAllByTestId("option-value");

    fireEvent.change(screen.getByTestId("username"));

    expect(screen.getByTestId("username")).toHaveAttribute("id", "username");
    expect(screen.getByTestId("username")).toBeInTheDocument();
    expect(options.length).toBe(2);

    await expect(onChangeHandlerMock).toHaveReturnedWith("Change triggered");
    expect(asFragment()).toMatchSnapshot();
  });

  afterAll(() => {
    jest.clearAllMocks();
  });
});
