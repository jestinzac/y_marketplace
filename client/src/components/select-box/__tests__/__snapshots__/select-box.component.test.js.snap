// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBox should render the element with values and trigger event on change 1`] = `
<DocumentFragment>
  <label
    class="pt-10 m-5 cursor-pointer"
    for="username"
  >
    Available Jobs for: 
  </label>
  <select
    class="shadow-inner mx-4 h-11"
    data-testid="username"
    id="username"
    name="username"
  >
    <option
      data-testid="option-value"
      value="1"
    >
      John
    </option>
    <option
      data-testid="option-value"
      value="2"
    >
      Smith
    </option>
  </select>
</DocumentFragment>
`;

exports[`SelectBox should render the element with values if passed 1`] = `
<DocumentFragment>
  <label
    class="pt-10 m-5 cursor-pointer"
    for="username"
  >
    Available Jobs for: 
  </label>
  <select
    class="shadow-inner mx-4 h-11"
    data-testid="username"
    id="username"
    name="username"
  >
    <option
      data-testid="option-value"
      value="1"
    >
      John
    </option>
    <option
      data-testid="option-value"
      value="2"
    >
      Smith
    </option>
  </select>
</DocumentFragment>
`;

exports[`SelectBox should render the element without any values 1`] = `
<DocumentFragment>
  <label
    class="pt-10 m-5 cursor-pointer"
    for="username"
  >
    Available Jobs for: 
  </label>
  <select
    class="shadow-inner mx-4 h-11"
    data-testid="username"
    id="username"
    name="username"
  />
</DocumentFragment>
`;
